### PR TITLE
Fix Validate Password filter

### LINKED ISSUE #308: Vaildate Password Filter only works in certain situations
The current implementation of `SymphonyMember::filter_ValidatePassword` works correctly if the field handles are `username`, `email` or `password`. However, it does not in other cases.

The main issue is that it uses `MemberSection::getFieldHandle` to build the array keys for the data array that is passed to `SymphonyMember::findMemberIDFromCredentials`. However, the latter expects an array with Member field types like either `username` or `email`, plus `password`.

I will send a PR.

### PR DESCRIPTION
**Pull request by michael-e**
Fixes #308

### CODE REVIEW

File: lib/member.symphony.php [hunk 1]
```diff
@@ -413,13 +413,15 @@ public function filter_ValidatePassword(array &$context) {
 				$password = $_POST['fields'][$this->section->getFieldHandle('authentication')]['validate'];
 
 				if ($password) {
-					// Handle which is the Identity field, either the Member: Username or Member: Email field
-					$identity = is_null($this->section->getFieldHandle('identity')) ? 'email' : 'identity';
+					// Handle which is the Identity field, either the Member: Username or Member: Email field;
+					// the data array passed to findMemberIDFromCredentials() needs either 'username' or
+					// 'email' as key, so we must not use the string 'identity' in this case.
+					$identity = is_null($this->section->getFieldHandle('identity')) ? 'email' : 'username';
```
Comment (nitriques) [inline lib/member.symphony.php]: Would ! be better ?

**Review by nitriques — APPROVED** [review]

File: lib/member.symphony.php (hunk 1 above)
Comment (michael-e) [reply lib/member.symphony.php]: This is adopted from here: https://github.com/symphonycms/members/blob/1.10.1/lib/member.symphony.php#L417. I wouldn't like to do it differently when re-using it.

Comment (nitriques) [reply lib/member.symphony.php]: Ok great.

**Comment by michael-e** [comment]
Will you make a bugfix release?

**Comment by michael-e** [comment]
Now that (https://github.com/symphonycms/members/releases/tag/1.10.2) was fast. Thanks!

**Pull request merged**